src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1899,46 +1913,98 @@ internal static void MakeSeparatorListAny(ReadOnlySpan<char> source, ReadOnlySpa
 
         private static void MakeSeparatorListVectorized(ReadOnlySpan<char> sourceSpan, ref ValueListBuilder<int> sepListBuilder, char c, char c2, char c3)
         {
-            // Redundant test so we won't prejit remainder of this method
-            // on platforms where it is not supported
-            if (!Vector128.IsHardwareAccelerated)
+            Debug.Assert(sourceSpan.Length >= Vector128<ushort>.Count);
+            nuint lengthToExamine = (uint)sourceSpan.Length;
+            nuint offset = 0;
+            ref char source = ref MemoryMarshal.GetReference(sourceSpan);
+
+            if (Vector512.IsHardwareAccelerated && lengthToExamine >= (uint)Vector512<ushort>.Count*2)
             {
-                throw new PlatformNotSupportedException();
-            }
+                Vector512<ushort> v1 = Vector512.Create((ushort)c);
+                Vector512<ushort> v2 = Vector512.Create((ushort)c2);
+                Vector512<ushort> v3 = Vector512.Create((ushort)c3);
 
-            Debug.Assert(sourceSpan.Length >= Vector128<ushort>.Count);
+                do
+                {
+                    Vector512<ushort> vector = Vector512.LoadUnsafe(ref source, offset);
+                    Vector512<ushort> v1Eq = Vector512.Equals(vector, v1);
+                    Vector512<ushort> v2Eq = Vector512.Equals(vector, v2);
+                    Vector512<ushort> v3Eq = Vector512.Equals(vector, v3);
+                    Vector512<byte> cmp = (v1Eq | v2Eq | v3Eq).AsByte();
```

Comment:
Does having these in locals hinder codegen at all?

We don't currently track whether a local is a mask, so doing this might result in better codegen (vpcmp, vpcmp, kor, vpcmp, kor):
```suggestion
Vector128<byte> cmp = (Vector512.Equals(vector, v1) | Vector512.Equals(vector, v2) | Vector512.Equals(vector, v3)).AsByte();
```

Reply:
Hi,

Yes both the above pieces of code have the same disassembly. 

```
G_M000_IG04:                ;; offset=0x0088
       vmovups  zmm0, zmmword ptr [r15+2*r14]
       vpcmpeqw k1, zmm0, zmm6
       vpcmpeqw k2, zmm0, zmm7
       kord     k1, k1, k2
       vpcmpeqw k2, zmm0, zmm8
       kord     k1, k1, k2
       vpmovm2w zmm0, k1
       vptestmb k1, zmm0, zmm0
       kortestq k1, k1
       je       G_M000_IG10
```

Let me know if you prefer one over the another due incase we are doing some optimization on kmask